File: LeetCode-Problems/Medium/XOR-Queries-of-Subarray.java
//Java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int[] rArr = new int[queries.length];
        int sum = 0;
        for (int i = 0; i < queries.length; i++){
            int start = queries[i][0];
            int end = queries[i][1];
            sum = arr[start];
            for (int j = start; j < end; j++){
                sum = (sum ^ arr[j+1]);
            }
            rArr[i] = sum;
        }
        return rArr;   
    }
}

File: LeetCode-Problems/Medium/XOR-Queries-of-Subarray.py
# Python
# Unfortunately this doesn't work as its Time Complexity is O(n^2), which ends up being TLE on Leetcode
class Solution(object):
    def xorQueries(self, arr, queries):
        """
        :type arr: List[int]
        :type queries: List[List[int]]
        :rtype: List[int]
        """
        rArr = list()
        rSum = 0
        for x in range(0, len(queries)):
            start = queries[x][0]
            end = queries[x][1]
            rSum = arr[start]
            for y in range(start, end):
                rSum = (rSum ^ arr[y+1])
            rArr.append(rSum)
        return rArr
        
# Alternative with less Time Complexity
class Solution(object):
    def xorQueries(self, arr, queries):
        """
        :type arr: List[int]
        :type queries: List[List[int]]
        :rtype: List[int]
        """
        rArr = list()
        leng = len(arr)
        xorTable = [0] * (leng+1)
        for x in range (0, leng):
            xorTable[x+1] = xorTable[x] ^ arr[x]
        for x in range (0, len(queries)):
            rArr.append(xorTable[queries[x][0]] ^ xorTable[(queries[x][1])+1])
        return rArr
            
